pkg/reconcilermanager/controllers/secret.go

```diff
@@ -51,78 +51,124 @@ func shouldUpsertPrivateCertSecret(rs *v1beta1.RepoSync) bool {
+// namespace.
+func upsertAuthSecret(ctx context.Context, rs *v1beta1.RepoSync, c client.Client, reconcilerRef types.NamespacedName) (client.ObjectKey, error) {
+	rsRef := client.ObjectKeyFromObject(rs)
+	switch {
+	case shouldUpsertGitSecret(rs):
+		nsSecretRef, cmsSecretRef := gitSecretRefs(rsRef, reconcilerRef, rs.Spec.Git.SecretRef.Name)
+		userSecret, err := getUserSecret(ctx, c, nsSecretRef)
 		if err != nil {
-			return err
+			return cmsSecretRef, errors.Wrap(err, "user secret required for git client authentication")
 		}
-	}
-	if shouldUpsertGitSecret(rs) {
-		err := upsertSecret(ctx, rs, c, reconcilerName, rs.Spec.Git.SecretRef.Name)
+		err = upsertSecret(ctx, c, cmsSecretRef, rsRef, userSecret)
+		if err != nil {
+			return cmsSecretRef, err
+		}
+		return cmsSecretRef, nil
+	case shouldUpsertHelmSecret(rs):
+		nsSecretRef, cmsSecretRef := gitSecretRefs(rsRef, reconcilerRef, rs.Spec.Helm.SecretRef.Name)
+		userSecret, err := getUserSecret(ctx, c, nsSecretRef)
 		if err != nil {
-			return err
+			return cmsSecretRef, errors.Wrap(err, "user secret required for helm client authentication")
 		}
+		err = upsertSecret(ctx, c, cmsSecretRef, rsRef, userSecret)
+		if err != nil {
+			return cmsSecretRef, err
+		}
+		return cmsSecretRef, nil
+	default:
+		// No secret required
+		return client.ObjectKey{}, nil
 	}
-	if shouldUpsertHelmSecret(rs) {
-		err := upsertSecret(ctx, rs, c, reconcilerName, rs.Spec.Helm.SecretRef.Name)
+}
+
+// upsertCACertSecret creates or updates the CA cert secret in the
+// config-management-system namespace using an existing secret in the RepoSync
+// namespace.
+func upsertCACertSecret(ctx context.Context, rs *v1beta1.RepoSync, c client.Client, reconcilerRef types.NamespacedName) (client.ObjectKey, error) {
+	rsRef := client.ObjectKeyFromObject(rs)
+	if shouldUpsertPrivateCertSecret(rs) {
+		nsSecretRef, cmsSecretRef := gitSecretRefs(rsRef, reconcilerRef, rs.Spec.Git.PrivateCertSecret.Name)
+		userSecret, err := getUserSecret(ctx, c, nsSecretRef)
 		if err != nil {
-			return err
+			return cmsSecretRef, errors.Wrap(err, "user secret required for git server validation")
 		}
+		err = upsertSecret(ctx, c, cmsSecretRef, rsRef, userSecret)
+		if err != nil {
+			return cmsSecretRef, err
+		}
+		return cmsSecretRef, nil
 	}
-	return nil
+	// No secret required
+	return client.ObjectKey{}, nil
 }
 
-// upsertSecret creates or updates a secret in config-management-system
-// namespace using an existing secret in the reposync.namespace.
-func upsertSecret(ctx context.Context, rs *v1beta1.RepoSync, c client.Client, reconcilerName, nsSecretName string) error {
-	// namespaceSecret represent secret in reposync.namespace.
-	namespaceSecret := &corev1.Secret{}
-	if err := get(ctx, nsSecretName, rs.Namespace, namespaceSecret, c); err != nil {
+func gitSecretRefs(rsRef, reconcilerRef client.ObjectKey, secretName string) (nsSecretRef, cmsSecretRef client.ObjectKey) {
```

Comment:
Did you intend this to be `getSecretRefs`?

Reply:
Fixed